google-fcm/src/test/scala/akka/stream/alpakka/google/firebase/fcm/impl/FcmSenderSpec.scala

```diff
@@ -0,0 +1,106 @@
+import akka.http.scaladsl.{HttpExt, HttpsConnectionContext}
+import akka.stream.ActorMaterializer
+import akka.stream.alpakka.google.firebase.fcm.FcmFlowModels.{FcmErrorResponse, FcmSend, FcmSuccessResponse}
+import akka.stream.alpakka.google.firebase.fcm.{FcmFlowModels, FcmNotification}
+import akka.testkit.TestKit
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{verify, when}
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+
+class FcmSenderSpec
+    extends TestKit(ActorSystem())
+    with WordSpecLike
+    with Matchers
+    with MockitoSugar
+    with BeforeAndAfterAll {
+
+  override def afterAll: Unit =
+    TestKit.shutdownActorSystem(system)
+  val timeout = 500.millis
+
+  implicit val materializer = ActorMaterializer()
+
+  "FcmSender" should {
+
+    "call the api as the docs want to" in {
+      val sender = new FcmSender
+      val http = mock[HttpExt]
+      when(
+        http.singleRequest(any[HttpRequest](),
+                           any[HttpsConnectionContext](),
+                           any[ConnectionPoolSettings](),
+                           any[LoggingAdapter]())
+      ).thenReturn(
+        Future.successful(HttpResponse(entity = HttpEntity(ContentTypes.`application/json`, """{"name": ""}""")))
+      )
+
+      Await.result(sender.send("projectId", "token", http, FcmSend(false, FcmNotification.empty)), timeout)
+
+      val captor: ArgumentCaptor[HttpRequest] = ArgumentCaptor.forClass(classOf[HttpRequest])
+      verify(http).singleRequest(captor.capture(),
+                                 any[HttpsConnectionContext](),
+                                 any[ConnectionPoolSettings](),
+                                 any[LoggingAdapter]())
+      val request: HttpRequest = captor.getValue
+      request.entity shouldBe HttpEntity(ContentTypes.`application/json`, """{"validate_only":false,"message":{}}""")
+      request.uri.toString shouldBe "https://fcm.googleapis.com/v1/projects/projectId/messages:send"
+      request.headers.size shouldBe 1
+      request.headers.head should matchPattern { case HttpHeader("authorization", "Bearer token") => }
+    }
+
+    "parse the success response correctly" in {
+      val sender = new FcmSender
+      val http = mock[HttpExt]
+      when(
+        http.singleRequest(any[HttpRequest](),
+                           any[HttpsConnectionContext](),
+                           any[ConnectionPoolSettings](),
+                           any[LoggingAdapter]())
+      ).thenReturn(
+        Future.successful(HttpResponse(entity = HttpEntity(ContentTypes.`application/json`, """{"name": "test"}""")))
+      )
+
+      val result: FcmFlowModels.FcmResponse =
+        Await.result(sender.send("projectId", "token", http, FcmSend(false, FcmNotification.empty)), timeout)
```

Comment:
You might want to use Scalatest's ScalaFutures trait instead of Await.

Reply:
This was not as successful as I want bcs of the multiple assertions.